lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dva/DvaThirdPartyDocumentGatewayTest.java

```diff
@@ -136,6 +141,27 @@ void shouldInvokeThirdPartyAPI()
                 httpRequestCaptor.getValue().getFirstHeader("Content-Type").getValue());
     }
 
+    @Test
+    void thirdPartyApiReturnsErrorOnHashValidation()
+            throws IOException, InterruptedException, CertificateException, ParseException,
+                    JOSEException, OAuthHttpResponseExceptionWithErrorBody,
+                    NoSuchAlgorithmException, InvalidKeySpecException {
+
+        DvaPayload dvaPayload = new DvaPayload();
+        dvaPayload.setSurname("Sur");
+        dvaPayload.setForenames(Arrays.asList("Fore"));
+
+        DvaResponse dvaResponse = new DvaResponse();
+        dvaResponse.setRequestHash(
+                "bd89175d1c1120864ba1c9a3ac856f4ad8ca16b1730d6755833d47a704114a50");
+        boolean isValidHash =
```

Comment:
Will this not fail as the validator is a null?

Reply:
I thought it was working as it gave the desired outcome in the assert.
But after checking with identical values, and printing them to confirm they are identical, you're right that it's just asserting false that `null` does not equal any string.